sequencer/statetransition.go

```diff
@@ -58,11 +58,19 @@ func (s *Sequencer) processPendingTransitions() {
 		// If the batch is nil, skip it
 		if batch == nil || len(batch.Ballots) == 0 {
 			log.Debugw("no ballots in batch", "batchID", batchID)
-			if err := s.stg.MarkBallotBatchFailed(batchID); err != nil {
+			if err := s.stg.MarkAggregatorBatchFailed(batchID); err != nil {
 				log.Errorw(err, "failed to mark ballot batch as failed")
 			}
 			return true // Continue to next process ID
 		}
+
+		// If there are pending txs, skip this process ID
+		if s.stg.HasPendingTx(storage.StateTransitionTx, pid) {
```

Comment:
It sounds weird that you are using `storage` con consult if there is a pending Tx. Wouldn't it be the txmanager the one who should have this information?